extensions/amp-accordion/1.0/amp-accordion.js

```diff
@@ -48,17 +51,28 @@ class AmpAccordion extends PreactBaseElement {
       api./*OK*/ collapse(invocation.args && invocation.args['section'])
     );
 
+    const action = Services.actionServiceForDoc(this.element);
+    const triggerEvent = (name, section, trust) => {
+      const event = createCustomEvent(
+        this.win,
+        `accordionSection.${name}`,
+        dict({})
+      );
+      action.trigger(section, name, event, trust);
+      this.element.dispatchCustomEvent(name);
+    };
+
     const {element} = this;
 
     const mu = new MutationObserver(() => {
-      this.mutateProps(getState(element, mu));
+      this.mutateProps(getState(element, mu, triggerEvent));
```

Comment:
IMHO it's not worth it to pass this function around. It can just be a module-level function here that would look up the action service and do the rest. Though it's not a very critical point.